tests/apps/test_graph_similarity.py

```diff
@@ -89,3 +92,113 @@ def test_sample_to_event(dim, max_count_per_mode):
     events = [similarity.sample_to_event(o, max_count_per_mode) for o in orbits]
 
     assert events == target_events
+
+
+class TestOrbitToSample:
+    """Tests for the function ``strawberryfields.apps.graph.similarity.orbit_to_sample``"""
+
+    def test_low_modes(self):
+        """Test if function raises a ``ValueError`` if fed an argument for ``modes`` that does
+        not exceed the length of the input orbit."""
+        with pytest.raises(ValueError, match="Number of modes cannot"):
```

Comment:
the `match` argument accepts a regex string, so it's even more powerful, you can place any regex there that you want to match!

Rule of thumb is to make it specific enough so that you will only catch the error you intend to.

Reply:
Yes, it's enough